### PR TITLE
Parse imdb.com URLs, add year support to search

### PR DESCRIPTION
**Pull request by kwaaak**
Mostly (all of it) stolen from @freeboson's https://github.com/sopel-irc/sopel/pull/1195
It's working on Python3.6 for me, haven't tested Python2.

### CODE REVIEW

**Comment by RustyBower** [comment]
Can you clean up the flake8 issues?

File: sopel_modules/imdb/imdb.py [hunk 1]
```diff
@@ -44,35 +51,74 @@ def imdb(bot, trigger):
...
         else:
-            LOGGER.warning(
-                'Got an error from the OMDb api, search phrase was %s; data was %s',
-                word, str(data))
+            LOGGER.info(
+                'Got an error from the OMDb api, search parameters were %s; data was %s',
+                str(params), str(data))
             message = '[Error] Got an error from OMDbapi'
     else:
+        data = data.json()
```
Comment (RustyBower) [inline sopel_modules/imdb/imdb.py]: We already define data = data.json(), you can nuke this line

File: sopel_modules/imdb/imdb.py [hunk 2]
```diff
@@ -44,35 +51,74 @@ def imdb(bot, trigger):
...
+            message += ' | IMDB Link: http://imdb.com/title/' + data['imdbID']
+
+        plot = data['Plot']
+        if len(message.format(plot)) > 300:
+            cliplen = 300 - (len(message) - 2 + 3) # remove {} add […]
+            plot = plot[:cliplen] + '[…]'
+        message = message.format(plot)
+    return message
+
+@sopel.module.rule('.*(imdb\.com\/title\/)(tt[0-9]+).*')
```
Comment (RustyBower) [inline sopel_modules/imdb/imdb.py]: Go ahead and import rule alongside commands and example and abbreviate this decorator to just @rule

File: sopel_modules/imdb/imdb.py [hunk 3]
```diff
@@ -30,11 +29,19 @@ def configure(config):
...
+def shutdown(bot):
+    del bot.memory['url_callbacks'][imdb_re]
 
-@commands('imdb')
+@commands('imdb', 'movie')
 @example('.imdb ThisTitleDoesNotExist', '[Error] Movie not found!')
-@example('.imdb Citizen Kane', '[Movie] Title: Citizen Kane | Year: 1941 | Rating: 8.4 | Genre: Drama, Mystery | IMDB Link: http://imdb.com/title/tt0033467')
-@example('.imdb Chuck', '[Series] Title: Chuck | Seasons: 5 | Year: 2007–2012 | Rating: 8.2 | Genre: Action, Comedy, Drama | IMDB Link: http://imdb.com/title/tt0934814')
+@example('.imdb Citizen Kane', '[Movie] Title: Citizen Kane | Year: 1941 | Rating: 8.3 | Tomatometer: 100% | Genre: Drama, Mystery | Plot: Following the death of a publishing tycoon, news reporters scramble to discover the meaning of his final utterance. | IMDB Link: http://imdb.com/title/tt0033467')
+@example('.imdb Chuck', '[Series] Title: Chuck | Seasons: 5 | Year: 2007–2012 | Rating: 8.2 | Genre: Action, Comedy, Drama | Plot: When a twenty-something computer geek inadvertently downloads critical government secrets into his brain, the C.I.A. and the N.S.A. assign two age[…] | IMDB Link: http://imdb.com/title/tt0934814')
```
Comment (RustyBower) [inline sopel_modules/imdb/imdb.py:49]: Can you add an example in here for a year? Like Halloween (2018) and Halloween (1978)

File: sopel_modules/imdb/imdb.py [hunk 4]
```diff
@@ -30,11 +29,19 @@ def configure(config):
         'Enter omdbapi.com API Key:'
     )
 
+def setup(bot):
+    bot.config.define_section('imdb', IMDBSection)    
+    if not bot.memory.contains('url_callbacks'):
+        bot.memory['url_callbacks'] = tools.SopelMemory()
```
Comment (RustyBower) [inline sopel_modules/imdb/imdb.py]: tools isn't imported

**Comment by kwaaak** [comment]
Thanks for the review, some obvious errors, I should have read the Travis log immediately

**Pull request merged**